### PR TITLE
feat(unlock-app): Add ability to save checkout config

### PR DESCRIPTION
**Pull request by searchableguy**
# Description

https://user-images.githubusercontent.com/73341821/226451404-18df12e6-2cfa-4a69-816a-df0a59cbd930.mp4

# Issues

Fixes #
Refs #

# Checklist:

- [ ] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

**Comment by searchableguy** [comment]
I think maybe better to just disallow people from creating same name config. 

That's a good UX point. I'm not sure the last updated information is relevant to users. We sort by last updated in the background.

File: locksmith/src/controllers/v2/checkoutController.ts [hunk 1]
```diff
@@ -52,14 +70,38 @@ export const getCheckoutConfigsByUser: RequestHandler = async (
...
+      id,
+      createdBy: userAddress,
+    },
+  })
+  if (checkoutConfig) {
+    await checkoutConfig.destroy()
+  }
+  return response.status(200).send({
+    message: 'Config deleted',
+  })
```
Comment (kalidiagne) [inline locksmith/src/controllers/v2/checkoutController.ts:106]:
```suggestion
  if (checkoutConfig) {
    await checkoutConfig.destroy()
     return response.status(200).send({
    message: 'Config deleted',
  })
  }
  return response.status(404).send({
    message: 'Not found',
  })
```

**Review by julien51 — APPROVED** [review]
I wonder if we could auto-save? I worry that users will "forget" to save...

File: locksmith/src/controllers/v2/checkoutController.ts (hunk 1 above)
Comment (kalidiagne) [reply locksmith/src/controllers/v2/checkoutController.ts:106]: here `'Config deleted'` is returned even if the config is not found

**Pull request merged**